### PR TITLE
squeeze functionality implemented

### PR DESCRIPTION
**Pull request by icyx000**
Purpose
Provide a backend service that allows us to collect user emails, store them in the database, and send a template to the user.

Requirements
 Develop a RESTful API endpoint to handle POST requests.
 Validate and verify incoming user data.
 Securely store user information in the database.
 Return appropriate responses.
 Handle errors and return appropriate status codes.
 Send a template to a specified email address using an email service (e.g., SendGrid, SMTP).
Acceptance Criteria
The page should be accessible at /api/v1/squeeze.
The endpoint should accept HTTP POST requests.
Squeeze Form Submission:

Endpoint: POST /api/v1/squeeze
Request Body:
{
  "email": "user@example.com",
  "first_name": "John",
  "last_name": "Doe",
  "phone": "08098761234",
  "location": "Lagos, Nigeria",
  "job_title": "Software Engineer",
  "company": "X-Corp",
  "interests": ["Web Development", "Cloud Computing"],
  "referral_source": "LinkedIn"
}
Validation: Ensure all fields are required and the email is in the correct format.
Response (Success):
{
  "message": "Your request has been received. You will get a template shortly."
}
Response (Error):
{
  "message": "Failed to submit your request",
  "status_code": 400
}
Tasks
 Create an endpoint (POST /api/v1/squeeze) to handle contact form submissions.
 Validate user input on the server side.
 Securely store user information in the database.
 Implement error handling and return appropriate status codes.
 Expected Outcome
A secure and efficient API endpoint that can receive user emails, store them in the database, and confirm the user's request.
The API endpoint allows users to submit a request via the "Squeeze" form and ensures the template is sent to the appropriate email address.

### CODE REVIEW

File: services/squeeze/squeeze.go [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+        FirstName:      req.FirstName,
+        LastName:       req.LastName,
+        Phone:          req.Phone,
+        Location:       req.Location,
+        JobTitle:       req.JobTitle,
+        Company:        req.Company,
+        Interests:      req.Interests,
+        ReferralSource: req.ReferralSource,
+    }
+    if err := storage.DB.Postgresql.Create(&userSubmission).Error; err != nil {
```
Comment (oxiginedev) [inline services/squeeze/squeeze.go]: All database operations (create, read, etc) should be a receiver function on the models. For this add a `CreateUserSubmission` method on the `UserSubmission` model

File: utility/email.go [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+func SendEmail(to, subject, body string) error {
+    m := gomail.NewMessage()
+
+    m.SetHeader("From", "your-email@example.com")
+    m.SetHeader("To", to)
+    m.SetHeader("Subject", subject)
+    m.SetBody("text/html", body)
+
+    d := gomail.NewDialer("smtp.example.com", 587, "your-email@example.com", "your-password")
```
Comment (oxiginedev) [inline utility/email.go]: Nice one, but these parameters should be read from environment

File: services/squeeze/squeeze.go [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+        return errors.New("failed to save submission to database")
+    }
+    return nil
+}
+
+
+func SendConfirmationEmail(email string) error {
```
Comment (oxiginedev) [inline services/squeeze/squeeze.go:32]: Let's find a way to send mails from templates, hard coding the email body is definitely not ideal.

File: internal/models/user_submission.go [hunk 4]
```diff
@@ -0,0 +1,17 @@
+package models
+
+import "time"
+
+type UserSubmission struct {
+    ID            uint      `gorm:"primaryKey" json:"id"`
```
Comment (oxiginedev) [inline internal/models/user_submission.go]: we are using UUID v7 for all primary keys

Comment (devffery) [reply internal/models/user_submission.go]: done

File: services/squeeze/squeeze.go (hunk 3 above)
Comment (devffery) [reply services/squeeze/squeeze.go:32]: you can review new code

**Pull request closed without merging**